src/Foundatio.Aliyun/Storage/AliyunFileStorage.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Aliyun.OSS;
+using Foundatio.Extensions;
+
+namespace Foundatio.Storage {
+    public class AliyunFileStorage : IFileStorage {
+        private readonly string _bucketName;
+        private readonly OssClient _client;
+
+        public AliyunFileStorage(string connectionString, string bucketName = "storage") {
+            var account = AliyunConnectionString.Parse(connectionString);
+            _client = account.CreateClient();
+            _bucketName = bucketName;
+            if (!_client.DoesBucketExist(_bucketName)) _client.CreateBucket(_bucketName);

```

Comment:
It's too bad they don't have a CreateBucketIfNotExist which just called Create Bucket and looked for a 409 or other response type.

Reply:
They do not support it, so we have to check the bucket exists before create.